### ISSUE TITLE
plexapi.library.section(<SECTION>).all() returns only two list elements of collection

### ISSUE THREAD

**Issue by awenny** [pkkid/python-plexapi#200]
I do have a quite extensive list of collections for some movies.
But if I look into the collections, I can only see a maximum of two per movie.

Eg:
```python
movie_library = plex.library.section("Movies")
for Movie in movie_library.all():
    movieCollections = []
    for X in Movie.collections:
        movieCollections.append(X.tag)
    print("{}: {}".format(Movie.title, len(movieCollections)))
```

In Plexweb, I can see all of them. BTW: which I have added via `addCollection([newCollection])` from plexapi.

I'm using plexapi 3.0.3 installed via pip on macOS and Python 3.6.3. PlexServer is most recent version running on Gentoo Linux.

**Comment by pkkid** [comment]
Does it work better if you add the line `Movie.reload()` in the code above?

```python
movie_library = plex.library.section("Movies")
for Movie in movie_library.all():
    Movie.reload()
    movieCollections = []
    for X in Movie.collections:
        movieCollections.append(X.tag)
    print("{}: {}".format(Movie.title, len(movieCollections)))
```

**Comment by Hellowlol** [comment]
The reload has to be done manually if not it would render plexapi useless as it would querying all my media would require over 25000 sync http calls. Maybe we should explain this better in the docs since it comes up from time to time.

**Comment by awenny** [comment]
Unfortunately, the `Movie.reload()` does not help.
I've even restarted the Plex server and looked into the original XML response from the API call - all collections are there. Also, my small python program is restarted after the change of the collection - so no cache issue on the client side. If there is no other cache, I need to refresh ....

Thanks, BTW, for the great work, providing this interface to the Plex API :)

**Comment by pkkid** [comment]
Can you enable logging on your setup?  You can do this by adding the following content to `~/.config/plexapi/config.ini` then watching the specified log file as you run your program. Specifically look for the URL requested when you call `movie_library.all()` and compare the XML from that URL to the actual full movie details that's requested when you call `Movie.reload()`. You should see that the `all` listing is a partial content. -- I wonder if .reload() is not actually overwriting the partial content correctly (which would definitely be a bug).

```ini
[log]
path = /path/to/plexapi.log
level = debug
show_secrets = true
```

**Comment by Hellowlol** [comment]
or by adding
````
import logging
logging.basicConfig(level=logging.DEBUG)
````

**Comment by Hellowlol** [comment]
@awenny can you post the url and the xml result. Remember to remove any tokens in the url and the response.

**Comment by awenny** [comment]
It's embarrassing, but I have to admit, I cannot reproduce the issue today. I've checked the version, I've used yesterday - thanks to Dropbox for versioning every saved version of the script. But today, it's working as you said. If I don't use the `Movie.reload()`, there are only two collections available. But if using it, then all collections are available.

I've also added
```
    if len(Movie.collections) == 2:
        Movie.reload()
```
to run the reload only if two collections are available. Then we don't know, if the movie has more than two collections and need to reload.
